File: packages/stera_recorder/ios/stera_recorder/Sources/stera_recorder/coordination/RecordingStateManager.swift
import Foundation

class RecordingStateManager {
    func canStart(currentState: ArRecordingState) -> Bool {
        return currentState == .ready
    }

    func canStop(currentState: ArRecordingState) -> Bool {
        return currentState == .recording || currentState == .paused
    }

    func canPause(currentState: ArRecordingState) -> Bool {
        return currentState == .recording
    }

    func canResume(currentState: ArRecordingState) -> Bool {
        return currentState == .paused
    }

    func canCancel(currentState: ArRecordingState) -> Bool {
        return currentState == .recording || currentState == .paused
    }
}

File: packages/stera_recorder/android/src/main/kotlin/open/fpvlabs/stera/ar_recorder/coordination/RecordingStateManager.kt
package open.fpvlabs.stera.ar_recorder.coordination

import open.fpvlabs.stera.ar_recorder.ArRecordingState

class RecordingStateManager {
    fun canStart(currentState: ArRecordingState): Boolean = currentState == ArRecordingState.READY
    fun canStop(currentState: ArRecordingState): Boolean =
        currentState == ArRecordingState.RECORDING || currentState == ArRecordingState.PAUSED
    fun canPause(currentState: ArRecordingState): Boolean = currentState == ArRecordingState.RECORDING
    fun canResume(currentState: ArRecordingState): Boolean = currentState == ArRecordingState.PAUSED
    fun canCancel(currentState: ArRecordingState): Boolean =
        currentState == ArRecordingState.RECORDING || currentState == ArRecordingState.PAUSED
}
